app/src/main/java/com/android/bookswap/data/source/network/BookFirestoreSource.kt

```diff
@@ -252,6 +259,156 @@ class BooksFirestoreSource(private val db: FirebaseFirestore) : BooksRepository
         result.exception?.let { e -> callback(Result.failure(e)) }
       }
     }
+    /**
+     * If the Firestore operation fails, this method logs the exception and invokes the callback
+     * with the failure result.
+     *
+     * @param e The exception that occurred during the Firestore operation.
+     */
+    fun moveBookToArchive(book: DataBook) {
+      val bookMap =
+          mapOf(
+              "uuid" to book.uuid.toString(),
+              "title" to book.title,
+              "author" to book.author,
+              "description" to book.description,
+              "rating" to book.rating,
+              "photo" to book.photo,
+              "language" to book.language.toString(),
+              "isbn" to book.isbn,
+              "genres" to book.genres.map { it.toString() },
+              "userId" to book.userId.toString(),
+              "archived" to true,
+              "exchange" to book.exchange)
+
+      performFirestoreOperation(
+          db.collection(collectionBooks_Archived).document(book.uuid.toString()).set(bookMap)) {
+              result ->
+            if (result.isSuccess) deleteBooks(book.uuid, book, callback)
+            callback(result)
```

Comment:
Here you call delete the book but you do not readd it to the other side, i.e you remove it from the non-archived archive and fail to retrieve it. Additionally there is no test for this function, it should be tested if only shoddily.

Reply:
This part actually add the book